catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/StdidcAttribute.java

```diff
@@ -0,0 +1,203 @@
+            OP_NUM_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, OP_NUM_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute PASS_ATTRIBUTE = new StdidcAttribute(PASS,
+            PASS_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, PASS_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute REPLAY_REGEN_ATTRIBUTE = new StdidcAttribute(REPLAY_REGEN,
+            REPLAY_REGEN_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, REPLAY_REGEN_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute REPRO_NUM_ATTRIBUTE = new StdidcAttribute(REPRO_NUM,
+            REPRO_NUM_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, REPRO_NUM_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute START_COLUMN_ATTRIBUTE = new StdidcAttribute(START_COLUMN,
+            START_COLUMN_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, START_COLUMN_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute START_ROW_ATTRIBUTE = new StdidcAttribute(START_ROW,
+            START_ROW_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, START_ROW_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute START_SEGMENT_ATTRIBUTE = new StdidcAttribute(START_SEGMENT,
+            START_SEGMENT_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, START_SEGMENT_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute END_COLUMN_ATTRIBUTE = new StdidcAttribute(END_COLUMN,
+            END_COLUMN_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, END_COLUMN_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute END_ROW_ATTRIBUTE = new StdidcAttribute(END_ROW,
+            END_ROW_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, END_ROW_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    static final StdidcAttribute END_SEGMENT_ATTRIBUTE = new StdidcAttribute(END_SEGMENT,
+            END_SEGMENT_SHORT_NAME,
+            tre -> TreUtility.convertToString(tre, END_SEGMENT_SHORT_NAME),
+            BasicTypes.STRING_TYPE);
+
+    private StdidcAttribute(String longName, String shortName,
+            Function<Tre, Serializable> accessorFunction, AttributeType attributeType) {
+        super(longName, shortName, accessorFunction, attributeType);
+        ATTRIBUTES.add(this);
+    }
+
+    public static List<NitfAttribute<Tre>> getAttributes() {
+        return Collections.unmodifiableList(ATTRIBUTES);
+    }
+
+    private static String getAlpha3CountryCode(String cc) {
+        if (cc == null) {
+            return cc;
+        }
+
+        try {
+            return new Locale(Locale.ENGLISH.getLanguage(), cc).getISO3Country();
+        } catch (MissingResourceException e) {
+            return cc;
```

Comment:
Maybe the variable name `cc` could be changed to `alpha2CountryCode` so that it's more obvious what conversion is going on.

I think the expected behavior in relation to the GeoCoderPlugin should maybe be documented. That is, why is the alpha2 country code returned here, instead of null,  in the case that the conversion fails? Also, a log statement like 
`LOGGER.debug("Failed to convert country code {} to alpha-3 format. Returning null", cc, e);` might be worthwhile.

Reply:
Updated variable name and added logging. Also added some javadoc. The reason we don't return null is because we don't want a plugin down the line to try and populate the location.country-code field which might contradict the nitf alpha2 code.